### DISCUSSION TITLE
Array value disappears

### DISCUSSION THREAD

**Discussion by cbhai** [livewire/livewire#3715] [1. Help]
I am struggling to fix this since last 24 hours. Cannot understand what is wrong here.

Inside livewire, mount method() I am populating $this->estimateRooms 

```
$this->estimateRooms = DB::table('estimate_details')
		 ->select([DB::raw('room_id'), (DB::raw('sum(total) as room_total'))])
                 ->where('estimate_id', '=' , $id)
                 ->groupBy(DB::raw('room_id') )
                 ->get();
```

Inside create.blade.php, I have a table

```
                        @foreach ($estimateRooms as $room)
                        <tr>
                            <td>
                                {{$room->room_id}}
                            </td>
                            <td>
                                {{$room->room_total}}
                            </td>
```

Page loads, all looks good. Table shows correct values. One second later, I get below error message :

 Trying to get property 'room_id' of non-object (View: C:\laragon\www\cbq4\resources\views\livewire\estimate\create.blade.php) 

I tried checking if value is empty(), but in that case, Table remains empty, never loads values.

**Comment by joshhanley** [comment]
@cbhai Livewire doesn't support rehydrating custom queries (with selects and group bys), so you will need to call `->toArray()` on your results and use data as arrays instead of objects.

So at the end of your query

```php
...
->get()
->toArray();
```

Then in your view use 
```blade
<td>
  {{ $room['room_id'] }}
</td>
```

Hope this helps!

**Reply by cbhai** [reply]
I added ->toArray() at the end of query. That didn't help.

Inside blade foreach loop, I still use   {{ $room->'room_id'}} 
You suggested to access values using   {{ $room['room_id'] }} , but I cannot do that I am still using  {{ $room->'room_id'}} 
In class definition, I have 
public $estimateRooms = []; 

Page loads with required value, I can see it. And seconds  later it throws error popup. Please see screenshot. If I click outside of error popup, popup goes away & page is there with all required values. (just sharing this incase this points to something)

This is such basic thing and I have almost spent 2nd day on it.

![error](https://user-images.githubusercontent.com/20470497/133253164-38d8f312-fa17-4126-9734-1bf5aac10a0a.png)

**Reply by joshhanley** [reply]
@cbhai if you are converting it to an array, you have to change you blade view to `{{ $room['room_id'] }}` as that is the array access method for PHP. Why can't you just change your blade view and use the array access method?

The main way to get objects to work is if they are an eloquent model, which is queried using a standard query (not using `DB::table`, as Livewire doesn't support objects coming from `DB::table`). So for example if you have `$this->user = User::first()`, Livewire can support that.

As you are using custom queries, you need to make sure you convert the results of your query to arrays and then change your blade views to using array access.

Hope this helps!

**Comment by denizozturk** [comment]
Hi, @joshhanley has given you information on correct usage.

I wouldn't recommend going out of correct usage, but you can do it this way if you insist on doing it. However, if the object is corrupted after the edits you make on the array, doing this again every time will reduce the readability of the code.
![lw_object](https://user-images.githubusercontent.com/34531750/133224614-a5b542d5-8156-49c0-bd52-f2860fa8d7a7.png)

If you want manipulate the values too much, I suggest you do this conversion inside the render method.

**Reply by cbhai** [reply]
Thank you for suggestion. 

Looks like this has nothing to do with groupby query etc. This is some different problem. I tried using a normal eloquent query (different model just for checking) Client::all() on this & it worked.

But doesnt work with anything else.

I just used your suggestion, created 

```class level
$tableArray = [];  
```	    

``` 
$this->tableArray = json_decode(collect([
	['room_id' => 11 , 'roomTotal' => 1000],
        ['room_id' => 12 , 'roomTotal' => 2000],
])->toJson());
```

Inside blade

```
@foreach ($tableArray as $room)
                        <tr>
                            <td>
                                {{$room->room_id}}
                            </td>
                            <td>
                                {{$room->roomTotal}}
                            </td>
```

I get same error. No groupby,  No DB Raw thing. Just array. Does not work. 

Time being, have solved problem using multiple eloquent model query & loops.
